include/seqan3/alphabet/adaptation/char.hpp

```diff
@@ -27,117 +27,53 @@
+ * \ingroup adaptation
+ * \{
  */
-template <typename char_type>
-//!\cond
-    requires detail::is_char_adaptation_v<char_type>
-//!\endcond
-struct underlying_char<char_type>
-{
-    //!\brief The same type as char_type.
-    using type = char_type;
-};
 
-/*!\brief Specialisation of seqan3::underlying_rank for char types.
+/*!\brief Return the number of values the char type can take.
  * \tparam char_type One of `char`, `char16_t`, `char32_t` or `wchar_t`.
- * \relates seqan3::char_adaptation_concept
- * \sa seqan3::alphabet_rank_t
+ * \param chr The parameters actual value is ignored.
+ * \returns The respective sizes (e.g. 256 for `char`).
  */
 template <typename char_type>
-//!\cond
-    requires detail::is_char_adaptation_v<char_type>
-//!\endcond
-struct underlying_rank<char_type>
+constexpr auto alphabet_size(char_type const & SEQAN3_DOXYGEN_ONLY(chr)) noexcept
+    requires ::seqan3::detail::is_char_adaptation_v<char_type>
 {
-    //!\brief An unsigned integer type of the same size as `char_type`.
-    using type = meta::at<detail::char_adaptations_rank_types,
-                          meta::find_index<detail::char_adaptations, std::remove_reference_t<char_type>>>;
-};
-
-// ------------------------------------------------------------------
-// value metafunctions
-// ------------------------------------------------------------------
-
-/*!\brief Specialisation of seqan3::alphabet_size that delegates for char types.
- * \tparam char_type One of `char`, `char16_t`, `char32_t` or `wchar_t`.
- * \relates seqan3::char_adaptation_concept
- * \sa seqan3::alphabet_size_v
- */
-template <typename char_type>
-//!\cond
-    requires detail::is_char_adaptation_v<char_type>
-//!\endcond
-struct alphabet_size<char_type> :
-    std::integral_constant<detail::min_viable_uint_t<detail::size_in_values_v<char_type>>,
-                           detail::size_in_values_v<char_type>>
-{};
-
-// ------------------------------------------------------------------
-// free functions
-// ------------------------------------------------------------------
+    return detail::min_viable_uint_t<detail::size_in_values_v<char_type>>{detail::size_in_values_v<char_type>};
+}
 
-/*!\name Free function wrappers for the char alphabet adaptation
- * \brief For `char`, `char16_t` and `char32_t` do conversion to/from uint types.
- * \ingroup adaptation
- * \relates seqan3::char_adaptation_concept
- * \{
- */
 /*!\brief Converting char to char is no-op (it will just return the value you pass in).
  * \tparam char_type One of `char`, `char16_t`, `char32_t` or `wchar_t`.
  * \param chr The alphabet letter that you wish to convert to char.
  * \returns The letter's value in the alphabet's rank type (usually `char`).
  */
 template <typename char_type>
-constexpr alphabet_char_t<char_type> to_char(char_type const chr) noexcept
-    requires detail::is_char_adaptation_v<char_type>
+constexpr char_type to_char(char_type const chr) noexcept
```

Comment:
This confused me for a bit.
The `alphabet_char_t` is gone because this is now only for `char` types and as such the `to_char` is a no-op, right?

Then it should also be documented as no-op?
(the 
```
 * \param chr The alphabet letter that you wish to convert to char.
 * \returns The letter's value in the alphabet's rank type (usually `char`).
```
Is not exactly correct then?
(If accepted, this also applies to the `uint` adaptations.)